numalogic/connectors/rds/_base.py

```diff
@@ -0,0 +1,131 @@
+import pandas as pd
+from numalogic.connectors.rds._config import DatabaseServiceProvider, RDSConfig
+from numalogic.connectors.utils.aws.boto3_client_manager import Boto3ClientManager
+import logging
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class RDSDataFetcher:
```

Comment:
we need this as it will hold common functions for all different types of RDS database types. for example mysql, postgres, marid there are some common functionalities like database token, execute query. so I created this as abstract class which is imported by MysqlFetcher to implement all required functions or use default functionality.

@ab93

Reply:
Hmm, maybe have a different name then? The `fetcher` name is confusing